Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs

```diff
@@ -29,14 +29,26 @@ public class CreateOrUpdateBusinessPartnerHandler
 
         public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
         {
-            var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.DopplerUser.BillingSystemId);
+            try
+            {
+                var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.DopplerUser.BillingSystemId);
 
-            var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
-            dequeuedTask = await sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
+                var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
+                dequeuedTask = await sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
 
-            return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
-                await CreateBusinessPartner(dequeuedTask, sapSystem) :
-                await UpdateBusinessPartner(dequeuedTask, sapSystem);
+                return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
+                    await CreateBusinessPartner(dequeuedTask, sapSystem) :
+                    await UpdateBusinessPartner(dequeuedTask, sapSystem);
+            }
+            catch (Exception ex)
+            {
+                return new SapTaskResult
```

Comment:
What do you think to add a logger in standard output? for example
`_logger.LogError(e, $"Failed to create or update the Business Partner for the user: {dequeuedTask.BusinessPartner.CardCode}. Error: {e.Message}"`

Reply:
Implement the logger as you suggested, I had to modify several tests to make it compile and the tests run successfully